fedimint-core/src/task.rs

```diff
@@ -446,3 +446,104 @@ mod imp {
         }
     }
 }
+
+/// async trait that use MaybeSend
```

Comment:
Can we add an explanation for why this is necessary? Will be more obvious for someone new to the codebase why these macros were created

Reply:
Will add more docs for MaybeSend. Explaining the reason.